sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -127,9 +127,14 @@ public override void VisitGenericName(GenericNameSyntax node)
 
         public override void VisitElementAccessExpression(ElementAccessExpressionSyntax node)
         {
-            var symbols = GetSymbols(node);
-            UsedSymbols.UnionWith(symbols);
-            TryStorePropertyAccess(node, symbols);
+            if (node.Expression.IsKind(SyntaxKind.ThisExpression)
+                || knownSymbolNames.Contains(node.Expression.GetIdentifier()?.Identifier.ValueText))
```

Comment:
_Just thinking loud:_ In case `GetIdentifier()` returns `null`: We will not move forward and we will not inspect the symbol. Unless there's `null` in `knownSymbolNames`.

That feels as inconsistent and little hidden from readability POV. Can you do an explicit condition on the null identifier?

Not inspecting the symbol should be fine, right? Stuff like `GetDictionary()[index]` or `array[index][index]` should be recursively visited and handled on the top level.

Reply:
Indeed:
- for `array[index][index]` we get null identifier but there will be a second call with `array[index]` which will have `array` as identifier. The behavior is OK since we are not concerned with usage of the first part.
- for `GetDictionary()[index]` is basically the same case null identifier here and `VisitIdentifierName` will take care of `GetDictionary`